### PR TITLE
Milestone 9: Authorization

### PR DESCRIPTION
**Pull request by jaflih**
### Milestone 9: Authorization

- Install CanCanCan in the project.
- Add a role column to the user's table. 
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -24,6 +24,17 @@ def show
...
+  def destroy
+    post = Post.find(params[:id])
+    post.likes.destroy_all
+    post.comments.destroy_all
+    post.destroy
+    post.author.posts_counter = post.author.posts_counter - 1
+    post.save
+    p user_path(params[:user_id])
+    redirect_to user_path(params[:user_id])
+  end
```
Comment (DeliceLydia) [inline app/controllers/posts_controller.rb:36]: - [ ] I recognized that when you delete a post or a comment it shows that it was deleted but on counts, they are still there check the example below I deleted the 3rd post but on count are still there, please address that:
pos

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -17,6 +17,13 @@ def create
     end
   end
 
+  def destroy
+    comment = Comment.find(params[:id])
+    comment.post.comments_counter = comment.post.comments_counter - 1
+    comment.destroy
+    redirect_to { user_posts(current_user) }
+  end
```
Comment (DeliceLydia) [inline app/controllers/comments_controller.rb:25]: - [ ] Same applied here I am able to delete a comment but it still exists on comments counts please address that

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @user = current_user
+    @user = User.find(params[:user_id])
     @posts = @user.posts.includes(:comments)
   end
```
Comment (DeliceLydia) [inline app/controllers/posts_controller.rb:5]: - [ ] Please Uses `CanCanCan` for Post authorization in `Post Controller`
``load_and_authorize_resource``

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Kudos on:
- Awesome README
- Following git workflow

## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (jaflih) [reply app/controllers/posts_controller.rb]: @DeliceLydia Thank you ! I solve it.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (jaflih) [reply app/controllers/comments_controller.rb]: @DeliceLydia Thank you! I solve it.

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (jaflih) [reply app/controllers/posts_controller.rb]: @DeliceLydia ok it's done.

**Review by bimbolabuari — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**